internal/core/component_creator.go

```diff
@@ -127,6 +146,14 @@ func (cc *componentCreator) create(
 	loadConfig bool,
 	hclCtx *hcl.EvalContext,
 ) (*Component, error) {
+	// We first get the labels. We use labels to determine the proper use
+	// plugin and other things so we have to grab these first before we do
+	// anything else.
+	hclCtx, labels, err := cc.labels(hclCtx, app)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Note for reviewers: the main reason I did it this way here is because I know we're going to introduce label filters and so on into choosing what plugin to use and various other things. So I set it up so that we already have the labels at the beginning of an operation (rather than say... figuring this out purely in the config package) so we can make those decisions for future features.